### ISSUE TITLE
Animation doesn't finish completely/last few frames are missing

### ISSUE THREAD

**Issue by Cellane** [airbnb/lottie-ios#212]
I'm trying to integrate Lottie into my project with a Bodymovin JSON file I got from our designer and while the animation starts as expected, it doesn't actually ever reach its final state – to my unexperienced eye, it looks like last 3-5 frames are missing and I'm not entirely sure why. When I check the demo HTML file Bodymovin generated, the animation plays and loops just fine which leads me to believe that the export went through just fine and the issue might be on Lottie's side.

I understand Lottie doesn't support the entirety of After Effects features but I'm also not sure how I could verify that an unsupported feature was found in the JSON file (and our designer swears he didn't use anything from the list mentioned in README file). When I tried to include the JSON file into the Lottie's example app, the same behaviour (animation not finishing) happened. Furthermore, it seemed that the animation was just… stuck for the last ~10% of its duration.

Attached below in the ZIP archive is the original After Effects file, exported JSON file and the demo HTML file from Bodymovin as well. I'd be very grateful if someone more experienced could help me determine where the issue lies and if there's an easy way to solve this on my side or whether it's something that has to be improved in Lottie first.

[filter-to-x-button.zip](https://github.com/airbnb/lottie-ios/files/912412/filter-to-x-button.zip)

**Comment by Weijay** [comment]
Same here. I downloaded [progree-success](http://www.lottiefiles.com/71-progress-success) from Lottiesfile.com to tried. However, when I do animation in my project, it just missing the last part of the animation.

**Comment by alaraKalama** [comment]
If your designer makes the rotation and position transformation directly on the rectangle shape transform properties (instead of the shape layer transform) it should work fine. I am uploading the changed project. 

[filter-to-x-button.zip](https://github.com/airbnb/lottie-ios/files/1019156/filter-to-x-button.zip)

**Comment by Cellane** [comment]
Thank you so much for your assistance! I'll pass the message back to our designers but unfortunately due to time constraints, I had to settle for different method – simple and stupid `UIImageView`'s `animationImages`.

**Comment by NemoAir** [comment]
My animation was about 2s duration,and loop three times use four image layers.
it seems like all layers played animation at the start frame, some should play at 1.0s,1.5s,but they all played at the start point.
so the animation look like stoped about 1s duration when next loop play.
I checked the code,all animation layers number、animation layer duration、image aseets are correct.
this ios library seems lack some AE effect support in compare with android library.
We had to wait next release version to fix this.
